lib/runtime/AssetPrefetchPreloadRuntimeModule.js

```diff
@@ -0,0 +1,65 @@
+/*
+	MIT License http://www.opensource.org/licenses/mit-license.php
+	Author Tobias Koppers @sokra
+*/
+
+"use strict";
+
+const RuntimeGlobals = require("../RuntimeGlobals");
+const RuntimeModule = require("../RuntimeModule");
+const Template = require("../Template");
+
+/** @typedef {import("../Compilation")} Compilation */
+
+class AssetPrefetchPreloadRuntimeModule extends RuntimeModule {
+	/**
+	 * @param {string} type "prefetch" or "preload"
+	 */
+	constructor(type) {
+		super(`asset ${type}`);
+		this._type = type;
+	}
+
+	/**
+	 * @returns {string | null} runtime code
+	 */
+	generate() {
+		const { compilation } = this;
+		if (!compilation) return null;
+		const { runtimeTemplate } = compilation;
+		const fn =
+			this._type === "prefetch"
+				? RuntimeGlobals.prefetchAsset
+				: RuntimeGlobals.preloadAsset;
+
+		return Template.asString([
+			`${fn} = ${runtimeTemplate.basicFunction("url, as, fetchPriority", [
+				"var link = document.createElement('link');",
+				this._type === "prefetch"
+					? "link.rel = 'prefetch';"
+					: "link.rel = 'preload';",
+				"if(as) link.as = as;",
+				"link.href = url;",
+				"if(fetchPriority) {",
+				Template.indent([
+					"link.fetchPriority = fetchPriority;",
+					"link.setAttribute('fetchpriority', fetchPriority);"
+				]),
+				"}",
+				// Add nonce if needed
+				compilation.outputOptions.crossOriginLoading
+					? Template.asString([
+							"if(__webpack_require__.nc) {",
```

Comment:
Use RuntimeGlobals.scriptNonce instead.

Reply:
fixed in 8daacd42f3494bd06ea1a801f31ec070c1a730c1